dumper/lib-dumper-spi/src/main/java/com/google/edwmigration/dumper/plugin/lib/dumper/spi/RangerDumpFormat.java

```diff
@@ -0,0 +1,1073 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.plugin.lib.dumper.spi;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonFormat.Shape;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.time.Instant;
+import javax.annotation.Nullable;
+
+public interface RangerDumpFormat {
+
+  ObjectMapper MAPPER =
```

Comment:
Can we have only one `MAPPER` (the other one is in `RangerClient`)?

Reply:
I've kept them distinct because the one in `RangerClient` is solely used to (de)serialize Ranger API payloads, whereas the one in `RangerDumpFormat` is specifically used for serialization into the zip file. Each has its own set of settings.

Probably they could be merged in one, but I think it makes sense to keep these separate as some settings are only meaningful in one context (for example, `FAIL_ON_UNKNOWN_PROPERTIES` makes sense only when reading from Ranger API).